lua/customroles/soulbound.lua

```diff
@@ -0,0 +1,611 @@
+        local num = net.ReadUInt(4)
+        if not ply:IsSoulbound() then return end
+
+        local id = ply:GetNWString("TTTSoulboundAbility" .. tostring(num), "")
+        if #id == 0 then return end
+
+        local ability = SOULBOUND.Abilities[id]
+        if not ability:Enabled() then return end
+
+        local target = ply:GetObserverMode() ~= OBS_MODE_ROAMING and ply:GetObserverTarget() or nil
+        if not ability:Condition(ply, target) then return end
+
+        ability:Use(ply, target)
+    end)
+
+    ----------------------
+    -- ABILITY PURCHASE --
+    ----------------------
+
+    net.Receive("TTT_SoulboundBuyAbility", function(len, ply)
+        local id = net.ReadString()
+        if not ply:IsSoulbound() then return end
+
+        local max = soulbound_max_abilities:GetInt()
+        for i = 1, max do
+            local slotId = ply:GetNWString("TTTSoulboundAbility" .. tostring(i), "")
+            if #slotId > 0 then continue end
+
+            ply:SetNWString("TTTSoulboundAbility" .. tostring(i), id)
+
+            local ability = SOULBOUND.Abilities[id]
+            ability:Bought(ply)
+            return
+        end
+        ply:PrintMessage(HUD_PRINTTALK, "You can't buy another ability!")
+    end)
+
+    -------------
+    -- CLEANUP --
+    -------------
+
+    hook.Add("TTTPrepareRound", "Soulbound_TTTPrepareRound", function()
+        for _, p in ipairs(player.GetAll()) do
+            for i = 1, 9 do
+                p:SetNWString("TTTSoulboundAbility" .. tostring(i), "")
+            end
+        end
+    end)
+
+    -----------------
+    -- ALIVE CHECK --
+    -----------------
+
+    hook.Add("TTTPlayerSpawnForRound", "Soulbound_TTTPlayerSpawnForRound", function(ply, dead_only)
+        if not IsPlayer(ply) then return end
+        if ply:IsSoulbound() then
+            for i = 1, 9 do
+                ply:SetNWString("TTTSoulboundAbility" .. tostring(i), "")
+            end
+            ply:SetRole(ROLE_TRAITOR)
+            SendFullStateUpdate()
+        end
+    end)
+end
+
+if CLIENT then
+    local client
+
+    ----------
+    -- SHOP --
+    ----------
+
+    local function CreateFavTable()
+        if not sql.TableExists("ttt_soulbound_fav") then
+            local query = "CREATE TABLE ttt_soulbound_fav (sid64 TEXT, ability_id TEXT)"
+            sql.Query(query)
+        end
+    end
+
+    local function AddFavorite(sid64, ability_id)
+        local query = "INSERT INTO ttt_soulbound_fav VALUES('" .. sid64 .. "','" .. ability_id .. "')"
+        sql.Query(query)
+    end
+
+    local function RemoveFavorite(sid64, ability_id)
+        local query = "DELETE FROM ttt_soulbound_fav WHERE sid64 = '" .. sid64 .. "' AND `ability_id` = '" .. ability_id .. "'"
+        sql.Query(query)
+    end
+
+    local function GetFavorites(sid64)
+        local query = "SELECT ability_id FROM ttt_soulbound_fav WHERE sid64 = '" .. sid64 .. "'"
+        local result = sql.Query(query)
```

Comment:
I know this is probably a copy/paste from the BEM favorites, but this can be returned directly